### ISSUE TITLE
please clarify instructions to create icon to app, or better, automate it

### ISSUE THREAD

**Issue by CouldBeThis** [newville/pyshortcuts#35]
Hi!

I think I am missing a tiny bit of background knowledge required to follow your instructions. 

Readme.md states:

> Of course, that command might be the sort of command you might want to be able to launch by clicking on a desktop shortcut. We have just the tool for that. This script (included as gui_bootstrap.py in the examples folder) will create a desktop shortcut that launches the pyshortcut GUI

..... where is the examples folder? And what is supposed to be done with this file once it's located?

I tried searching to no avail, but sometimes Spotlight doesn't see non-user files. 

there is also this instruction which I think is related

```zsh
~> python setup.py install
~> python examples/gui_bootstrap.py
```

but surely you need to be in some specific directory to run something called setup.py? I can't figure out where.

I would suggest it would be *extremely handy* to create an argument like `--put-pyshortcuts-on-desktop` to handle this.

(Wow I never had occasion to think of a flag for anything before. It's hard! My first 10 ideas had letters already taken by other arguments. I'm sure you can think of a better idea though.)

thank you! 

\- from I am a person who is smart want to use python scripts, but not smart enough to make or fix them myself.

**Comment by newville** [comment]
> Readme.md states:
>> Of course, that command might be the sort of command you might want to be able to launch by clicking on a desktop shortcut. We have just the tool for that. This script (included as gui_bootstrap.py in the examples folder) will create a desktop shortcut that launches the pyshortcut GUI
> ..... where is the examples folder? And what is supposed to be done with this file once it's located?


There is an `examples` folder in the same source level directory as the `setup.py` script.  Or: https://github.com/newville/pyshortcuts/tree/master/examples

> there is also this instruction which I think is related
>
> ~> python setup.py install
> ~> python examples/gui_bootstrap.py
> but surely you need to be in some specific directory to run something called setup.py? I can't figure out where.

Yes, the source code directory, as would happen if you downloaded and unpacked the source code.  

> I would suggest it would be extremely handy to create an argument like --put-pyshortcuts-on-desktop to handle this.

Well, many people will want to use this library programmatically, say as part of the installation process for software they are writing (say, a script to "download", "install", "make shortcut").

> - from I am a person who is smart want to use python scripts, but not smart enough to make or fix them myself.

One of the beauties of python is that if you want to use python, you can figure it out.  Another beauty of python is that if you can use python, you can also start fixing scripts.

**Comment by CouldBeThis** [comment]
Thank you for your time. I do not think I am following you though. 

Here is  what I understood:

- The recommend way to install this is by using `pip install pyshortcuts`; which will make everything available via the CLI
- _But_ to make an icon on the desktop to access the app, it is _also_ required to clone the github repo and run something separately

Which, considering this is a script to make icons on the desktop, sounds wrong. 

At any rate it can be run from the cli so once I figure out how to make it function it shouldn't be impossible to make the icon. 

I want to use this on other people's work to make it easier for me to use, and in the event that it was possible (I am not sure if it would be), to contribute back.

**Comment by newville** [comment]
@CouldBeThis 
> Here is what I understood:
> 
> The recommend way to install this is by using pip install pyshortcuts; which will make everything available via the CLI

Yes.

> But to make an icon on the desktop to access the app, it is also required to clone the github repo and run something
> separately

Nope.  

> Which, considering this is a script to make icons on the desktop, sounds wrong.

> I want to use this on other people's work to make it easier for me to use, and in the event that it was possible (I am not sure if it would be), to contribute back.

Does running `pyshortcuts` from the command-line not work for you? The docs tell you how to use the program.  

At the end of the doc, it also gives the text of a script (that script is also in the `examples` folder of the source code, which is what you originally asked:  "where is that file?").   Take that text, save it as file called `bootstrap_pyshortcuts.py` or whatever you want.  Running that `python bootsrap_pyshortcuts.py` will put a shortcut on your desktop that will run the GUI version of pyshortcuts that will help you create shortcuts (you'll need wxpython installed to run this).   You don't need to use the GUI version, but it can be helpful for testing and debugging.

The text of that script also shows you how to make a shortcut from a python script.

I hope that is clear.

**Comment by CouldBeThis** [comment]
Hi, thank you I have done this. 

```
python bootstrap_pyshortcuts.py
```

this does create a file on the desktop called `PyShortcut`. When clicked it has the effect of making a tiny movement in the dock like something was going to start opening but closed very quickly. 

Opened the contents. Looks like the meat is in `~/Desktop/PyShortcut.app/Contents/MacOS/PyShortcut` 

```python
#!/bin/bash
## Make sure to set PYTHONEXECUTABLE to Python that created this script
export PYTHONEXECUTABLE=/usr/local/Cellar/python@2/2.7.16_1/Frameworks/Python.framework/Versions/2.7/bin/python
export EXE=/usr/local/opt/python@2/bin/python2.7
export SCRIPT=/usr/local/Cellar/python@2/2.7.16_1/Frameworks/Python.framework/Versions/2.7/bin/pyshortcut
export ARGS='--wxgui'
$EXE $SCRIPT $ARGS

```

On the other hand the is helpful because I can tell what sort of information the boxes in `pyshortcut -w` are looking for. So tried this:
![shortcut](https://user-images.githubusercontent.com/53547181/101123344-a9956580-35c2-11eb-9eb9-0c3f28e4f184.png)

`create shortcut` has no effect. 

 `save python code` creates a file and outputs in the terminal. Terminal:

```python
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/pyshortcuts/wxgui.py", line 306, in onCreate
    make_shortcut(script, **opts)
  File "/usr/local/lib/python2.7/site-packages/pyshortcuts/darwin.py", line 83, in make_shortcut
    folder=folder, icon=icon)
  File "/usr/local/lib/python2.7/site-packages/pyshortcuts/shortcut.py", line 66, in shortcut
    raise ValueError("`script` for shortcut must be a non-zero length string")
ValueError: `script` for shortcut must be a non-zero length string

```

File `make_shortcut.py`:

```python
#!/usr/bin/env python
from pyshortcuts import make_shortcut
make_shortcut('/usr/local/Cellar/python@2/2.7.16_1/Frameworks/Python.framework/Versions/2.7/bin/pyshortcut --wxgui',
              name='this opens pyshortcut',
              description='pie crust from mix',
              folder=,
              icon=,
              terminal=False, desktop=True, startmenu=True,
              executable='/usr/local/opt/python@2/bin/python2.7')
```

**Comment by newville** [comment]
@CouldBeThis 

> this does create a file on the desktop called PyShortcut. 

OK, good - that says that a lot of things are working.

> When clicked it has the effect of making a tiny movement in the dock like something was going to start opening but closed very quickly.

Hm, I'm not sure why that would happen.  I would have guessed that wxpython or something else was not installed correctly.  But, if running that script by hand works (and you actually see the GUI), then I am much less certain.   I will say that I don't test with Python 2.7 and you should not be using it. 

> create shortcut has no effect.
> save python code creates a file and outputs in the terminal. Terminal:


> Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/pyshortcuts/wxgui.py", line 306, in onCreate
    make_shortcut(script, **opts)
  File "/usr/local/lib/python2.7/site-packages/pyshortcuts/darwin.py", line 83, in make_shortcut
    folder=folder, icon=icon)
  File "/usr/local/lib/python2.7/site-packages/pyshortcuts/shortcut.py", line 66, in shortcut
    raise ValueError("`script` for shortcut must be a non-zero length string")
ValueError: `script` for shortcut must be a non-zero length string

Ah, maybe those are related... I don't see that, but this:

```
#!/usr/bin/env python
from pyshortcuts import make_shortcut
make_shortcut('/usr/local/Cellar/python@2/2.7.16_1/Frameworks/Python.framework/Versions/2.7/bin/pyshortcut --wxgui',
              name='this opens pyshortcut',
              description='pie crust from mix',
              folder=,
              icon=,
              terminal=False, desktop=True, startmenu=True,
              executable='/usr/local/opt/python@2/bin/python2.7')

```

definitely indicates a problem: that `folder=, icon=, ` there are syntax errors.  I would bet that if you removed those that running the script might work.  But clearly, the GUI is generating a bad script at least in some cases (and that is really its only task, so you should be able to save/edit/modify that script to generate shortcuts.

I can't promise when I'll find time to investigate this bad code generated by the GUI, but if you figure it out, please pass on whatever you find or make a PR.  Thanks and sorry for the trouble.

**Comment by newville** [comment]
@CouldBeThis I think this is now fixed in 1.8.0, which also add ta `--bootstrap` (or just `-b`) option to `pyshortcut`.  This will "bootstrap" pyshortcut by creating a desktop shortcut for `pyshortcut --wxgui`.  

When you get a chance, please try that.  If you still have problems, please open a new issue.
